source/community/reactnative/src/components/actions/action-button.js

```diff
@@ -124,8 +125,24 @@ export class ActionButton extends React.Component {
 			else
 				mergedObject["actionData"] = this.data;
 		}
+		return mergedObject
+	}
+
+	/**
+	 * @description Invoked for the action type Constants.ActionSubmit
+	 */
+	onSubmitActionCalled() {
 		const { type, title = "", ignoreInputValidation } = this.payload;
-		let actionObject = { "type": type, "title": title, "data": mergedObject };
+		let actionObject = { "type": type, "title": title, "data": this.getMergeObject() };
+		this.onExecuteAction(actionObject, ignoreInputValidation);
+	}
+
+	/**
+	 * @description Invoked for the action type Constants.ActionExecute
+	 */
+	onExecuteActionCalled() {
+		const { type, verb = "", title = "", ignoreInputValidation } = this.payload;
```

Comment:
Can we refactor this as a single method? Maybe we can use a `boolean` to identify the **`ExecuteAction`** and add the `verb` to `actionObject`.

Reply:
Resolved